vendor/modules.txt

```diff
@@ -596,9 +596,10 @@ github.com/hashicorp/go-azure-sdk/resource-manager/logz/2020-10-01/tagrules
 github.com/hashicorp/go-azure-sdk/resource-manager/machinelearningservices/2023-10-01/datastore
 github.com/hashicorp/go-azure-sdk/resource-manager/machinelearningservices/2023-10-01/machinelearningcomputes
 github.com/hashicorp/go-azure-sdk/resource-manager/machinelearningservices/2023-10-01/workspaces
-github.com/hashicorp/go-azure-sdk/resource-manager/maintenance/2022-07-01-preview/configurationassignments
-github.com/hashicorp/go-azure-sdk/resource-manager/maintenance/2022-07-01-preview/maintenanceconfigurations
 github.com/hashicorp/go-azure-sdk/resource-manager/maintenance/2022-07-01-preview/publicmaintenanceconfigurations
```

Comment:
Looks like there's still 3 references to this older API Version in the codebase:

* `internal/services/mssqlmanagedinstance/mssql_managed_instance_resource.go`
* `internal/services/mssql/mssql_elasticpool_resource.go`
* `internal/services/mssql/mssql_database_resource.go`

As such can we update those too, which'll mean that we can remove this older SDK version entirely?